ansible/roles/controller/tasks/main.yml

```diff
@@ -0,0 +1,32 @@
+---
+# This role will install Controller in group 'controllers' in the environment inventory
+
+- name: pull the lastest image of controller
+  shell: "docker pull {{ docker_registry }}whisk/controller:{{ docker_image_tag }}"
+  when: docker_registry != ""
+
+- name: (re)start controller
+  docker:
+    name: controller
+    image: "{{ docker_registry }}whisk/controller:{{ docker_image_tag }}"
+    state: reloaded
+    restart_policy: "{{ docker.restart.policy }}"
+    log_driver: syslog
+    env:
+      "COMPONENT_NAME": "controller"
+      "CONSULSERVER_HOST": "{{ groups['consul_servers']|first }}"
```

Comment:
this should work if we put localhost. since consul agent runs on all machines now. the consul server is the local machine from the perspective of each machine. it does not have to know the real consul host anymore.